operations/reservations.md

```diff
@@ -267,6 +274,13 @@ Returns all reservations specified by any identifier, customer or other filter.
 | `Type` | string [Order note type](#order-note-type) | required | Type of the note. |
 | `CreatedUtc` | string | required | Creation date and time of the note in UTC timezone in ISO 8601 format. |
 
+#### Age category person count
+
+| Property | Type | Contract | Description |
+| --- | --- | --- | --- |
+| `AgeCategoryId` | string | required | Unique identifier of the age category. |
```

Comment:
missing link to the age category entity.

Reply:
For some reason there is no ageCategories/getAll yet documented. I'll add that in the separate PR and fix the link here, if that's fine by you.

Reply:
Aside from the question of what words to use, I'm a little concerned that we are about to make life a lot harder for the API user (not just them but also anyone that has to support this, e.g. looking at logs in customer support). If we deprecate 'AdultCount' and 'ChildCount' then we will immediately break all their existing code. Then in the new proposal, there is no easy way to get something as simple as the count of guests from the reservation record, as a user you will now not only have to step through an array but also do a further lookup using a separate endpoint. Can we not make it a bit easier for them? For example, (1) don't deprecate the two counts (just define them using a separate fixed logic like under 18 = child), (2) put the non-zero age categories into the array so they don't have to do a separate lookup. I know you want to mimic the DB / object architecture, but should that be at the expense of usability?

Reply:
@MikeAdamsMews How we will break their code if it's done in a nonbreaking way? The old approach is still supported, so there is no impact :)

Reply:
The way I am reading the diff it looks like 'AdultCount' and 'ChildCount' are being removed from the record, have I got that wrong?

Reply:
Also, though it's removed from the Api documentation, it's not removed from the endpoint response. We just remove it from the documentation so the old implementation is not used and is not getting adapted when someone new implements the endpoint.

Reply:
We will still be returning  'AdultCount' and 'ChildCount' it just removed from documentation.

Reply:
I see, thanks @vitezp for the clarification, although personally I don't like that approach - it means the documentation doesn't match the reality. I would rather keep it in and say 'deprecated parameters' or something like that.

Reply:
@MikeAdamsMews this is not the first time, we discuss this approach at least 4 times.

Reply:
@vitasamek have I really raised it that many times? Well, I guess I will keep on until you get the hint ;) The more I think about it, the more I think it is flat out bad practice. The purpose of documentation is to illuminate not obfuscate. We should be transparent and not hide parameters.

Reply:
@MikeAdamsMews This is a current way how we do documentation. We can change it in many ways :) But there is a huge difference between talking and making things happen. :))) Unfortunately, we're only talking...

Reply:
If there is no more questions/comments to this PR I'd like to move this further. We can discuss significant changes in the API offline not over this PR.

Reply:
@vitezp There is a comment, we cannot merge it because there is missing the important part - get all categories.